src/app/tests/TestReportScheduler.cpp

```diff
@@ -387,7 +457,351 @@ class TestReportScheduler
+        sTestTimerSynchronizedDelegate.IncrementMockTimestamp(System::Clock::Milliseconds64(1000));
+
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler1));
+        readHandler2->mObserver->OnBecameReportable(readHandler2);
+
+        // confirm report scheduled now
+        NL_TEST_ASSERT(aSuite,
+                       syncScheduler.mTestNextReportTimestamp == sTestTimerSynchronizedDelegate.GetCurrentMonotonicTimestamp());
+        // Increment the timestamp by 0 here to trigger an engine run as the mock timer is only calling the timeout callback if we
+        // increment the mock timestamp
+        sTestTimerSynchronizedDelegate.IncrementMockTimestamp(System::Clock::Milliseconds64(0));
+
+        // since the min interval on readHandler1 is 0, it should also be reportable now by sync mechanism
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler1));
+        NL_TEST_ASSERT(aSuite, node1->GetSyncTimestamp() == node2->GetMinTimestamp());
+
+        // Confirm that the next report emission is scheduled on the min timestamp of readHandler2 as it is the highest reportable
+        NL_TEST_ASSERT(aSuite, syncScheduler.mTestNextReportTimestamp == node2->GetMinTimestamp());
+
+        // Simulate a report emission for readHandler1
+        readHandler1->mObserver->OnSubscriptionAction(readHandler1);
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler1));
+
+        // ReadHandler 2 should still be reportable since it hasn't emitted a report yet
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler2));
+        readHandler2->ClearForceDirtyFlag();
+        readHandler2->mObserver->OnSubscriptionAction(readHandler2);
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler2));
+
+        // Validate next report scheduled on the max timestamp of readHandler1
+        NL_TEST_ASSERT(aSuite, node1->GetMaxTimestamp() > sTestTimerSynchronizedDelegate.GetCurrentMonotonicTimestamp());
+        NL_TEST_ASSERT(aSuite, syncScheduler.mTestNextReportTimestamp == node1->GetMaxTimestamp());
+
+        // Simulate readHandler1 becoming dirty after less than 1 seconds
+        sTestTimerSynchronizedDelegate.IncrementMockTimestamp(System::Clock::Milliseconds64(900));
+        readHandler1->ForceDirtyState();
+        NL_TEST_ASSERT(aSuite, syncScheduler.IsReportableNow(readHandler1));
+        NL_TEST_ASSERT(aSuite, !syncScheduler.IsReportableNow(readHandler2));
+        readHandler1->mObserver->OnBecameReportable(readHandler1);
+
+        // Validate next report scheduled on the min timestamp of readHandler1 (readHandler 2 is not currently reportable)
+        NL_TEST_ASSERT(aSuite,
+                       syncScheduler.mTestNextReportTimestamp == sTestTimerSynchronizedDelegate.GetCurrentMonotonicTimestamp());
```

Comment:
not sure about this. Shouldn't we have to wait till MinInterval of readHandler2 to sync both?

Reply:
This is to prevent long min handlers to hold back responsive handlers. Say the numbers were in hours and not in seconds, handler1 would have to wait 1hour before reporting, which would change the user experience on that platform quite a lot. 

Current compromise is not perfect either so this will have to be discussed before 1.3